lib/avro/builder/dsl.rb

```diff
@@ -111,9 +98,19 @@ def add_schema_object(object)
         schema_objects[object.fullname] = object if object.namespace
       end
 
+      def type(name, type_name, options = {}, &block)
```

Comment:
Should we give this a more command oriented name since it has moved out of the public DSL interface e.g. `create_type`?